etl/variable_mapping_translate.py

```diff
@@ -0,0 +1,291 @@
+        json.dump(mapping, f, indent=2)
+
+
+def _read_mapping(mapping_file: str) -> Dict[str, str]:
+    """Read mapping dictionary from disk (JSON file)."""
+    with open(mapping_file, "r") as f:
+        mapping: Dict[str, str] = json.load(f)
+    return mapping
+
+
+def _read_vars_from_env(path: str) -> Dict[str, Any]:
+    """Read environment variables from .env file."""
+    config = dotenv_values(path)
+    return {
+        "db": config["DB_NAME"],
+        "host": config["DB_HOST"],
+        "port": int(str(config["DB_PORT"])),
+        "user": config["DB_USER"],
+        "password": config.get("DB_PASS", ""),
+    }
+
+
+def _build_engine(conf: Dict[str, str]) -> Engine:
+    """Build SQL connection object"""
+    return create_engine("mysql://{user}:{password}@{host}:{port}/{db}?charset=utf8mb4".format(**conf))
+
+
+def variable_mapping_translate(conn_1: Engine, conn_2: Engine, mapping: Dict[str, str]) -> Dict[str, str]:
+    """Obtain equivalent variable mapping from old to new DB.
+
+    Parameters
+    ----------
+        conn_1: Engine)
+            Connection to database 1 (aka old DB).
+        conn_2: Engine)
+            Connection to database 2 (aka new DB).
+        mapping: dict
+            Dictionary with variable IDs mapping (according to database 1)
+    Returns
+    -------
+        dict:
+            Dictionary with variable IDs mapping according to database 2.
+    """
+    # Get info from connection 1
+    log.info("Getting variable details from database 1...")
+    df = build_df_from_mapping(conn_1, mapping)
+    # Get ids from connection 2
+    log.info("Get IDs from database 2...")
+    mapping_new = build_mapping_from_df(conn_2, df)
+    return mapping_new
+
+
+def build_df_from_mapping(conn: Engine, mapping: Dict[str, str]) -> pd.DataFrame:
+    """Build dataframe from variable ids.
+
+    Contains variable id, variable name and dataset name for the old and new variables.
+
+    Parameters
+    ----------
+    mapping: dict
+        Dictionary mapping old to new variable ids.
+
+    Returns
+    -------
+    pandas.DataFrame:
+        Dataframe with all variable info.
+    """
+    df_old, df_new = _build_dfs(conn, mapping)
+    df = _merge_dfs(df_old, df_new, mapping)
+    return df
+
+
+def _run_query_mapping_to_df(conn: Engine, variable_ids: Tuple[str, ...]) -> pd.DataFrame:
+    """Get complete variable df from variable ids.
+
+    Parameters
+    ----------
+    conn : Engine
+        DB connection object.
+    variable_ids: list
+        List with variable ids.
+
+    Returns
+    -------
+    pandas.DataFrame :
+        Dataframe with variable fields: `id`, `name` and `dataset_name`.
+    """
+    query = f"""
+        select variables.id id, variables.name name, datasets.name dataset_name
+        from variables
+        left join datasets on variables.datasetId=datasets.id
+        where variables.id in {variable_ids};
+    """
+    df: pd.DataFrame = pd.read_sql_query(_format_query(query), conn)
```

Comment:
You could use `params=` like this instead of `_format_query`

```
query = f"""
    select variables.id id, variables.name name, datasets.name dataset_name
    from variables
    left join datasets on variables.datasetId=datasets.id
    where variables.id in %(variable_ids)s;
"""
df: pd.DataFrame = pd.read_sql_query(query, conn, params={"variable_ids": variable_ids})
```

using `params` prevents SQL injection and might be easier to use (though honestly I don't think we'll be injecting ourselves...).

Reply:
Thanks, I'm modifying query calls!